### PR TITLE
Support the latest Cirq version (0.15.0)

### PR DESCRIPTION
**Pull request by andreamari**
Description
-----------

Apply changes such that Mitiq is compatible with the latest Cirq version.
Main changes are:

- Notation of Pauli operators. E.g. using `Z(q(1))` instead of `Z(0)`.
- Update of `mitiq.utils._equal()` function. A function to test if two circuits are equal. Now the native `circuit_A == circuit_B` is used instead of comparing DAG (directed acyclic graphs) objects. This means that two circuits with the same gates but different moment structure are now considered considered as different. So, in some tests, the function `_squash_moments()` is used to before comparing two circuits. In practice using `_squash_moments()` + the new `_equal` function is equivalent to using the previous (more tolerant)  `_equal` function.

License
-------

- [x] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

### CODE REVIEW

File: mitiq/zne/scaling/tests/test_folding.py [hunk 1]
```diff
@@ -1831,12 +1835,12 @@ def test_apply_fold_mask():
     assert _equal(folded, correct)
 
     folded = _apply_fold_mask(circ, [0, 3, 0, 0, 0, 0])
-    correct = Circuit([ops.H.on(qreg[1])] * 6) + circ
-    assert _equal(folded, correct)
+    correct = Circuit(*[ops.H.on(qreg[1])] * 6) + circ
```
Comment (natestemen) [inline mitiq/zne/scaling/tests/test_folding.py:1838]: Is the unpacking `*` needed here? If so, what for?

File: mitiq/observable/pauli.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def __init__(
 
     @property
     def coeff(self) -> complex:
-        return self._pauli.coefficient
+        return self._pauli.coefficient  # type: ignore
```
Comment (natestemen) [inline mitiq/observable/pauli.py:87]: Why is this needed now?

**Review by natestemen — APPROVED** [review]
Looks great Andrea! Thanks for getting this done so quickly. Just two questions, but nothing that prevents merging.

File: mitiq/zne/scaling/tests/test_folding.py (hunk 1 above)
Comment (andreamari) [reply mitiq/zne/scaling/tests/test_folding.py]: It's not actually. Removed in [2fdc9ce](https://github.com/unitaryfund/mitiq/pull/1379/commits/2fdc9cec0fe904d046c3ce476fd8f2072aa5eda2)

File: mitiq/observable/pauli.py (hunk 2 above)
Comment (andreamari) [reply mitiq/observable/pauli.py:87]: I guess `cirq.PauliString` now supports symbolic coefficients beyond complex coefficients. In Mitiq instead we only support /assume complex coefficients. `mypy` don't like this fact.
